appsensor-core/src/main/java/org/owasp/appsensor/core/Attack.java

```diff
@@ -30,14 +29,14 @@
  * @author John Melton (jtmelton@gmail.com) http://www.jtmelton.com/
  */
 @Entity
-public class Attack implements Serializable {
+public class Attack implements IAppsensorEntity {
 
 	private static final long serialVersionUID = 7231666413877649836L;
 
 	@Id
-	@Column
+	@Column(columnDefinition = "integer")
```

Comment:
@jtmelton This change is somewhat a "hack" to provide compatibility with ES. The problem I had with all entity classes was that I wanted to use them with ES. However, ES does not have support for numeric IDs. Instead, this is what a typical auto-generated ES id looks like: "AVYM6F-ZkMSDTGOdOLf0"

By changing the java type to String but explicitly defining the SQL database type as integer, already running SQL installations should stay compatible. Code changes will be necessary though, if getId() / setId() are used. 

On a side note: The MongoDB implementation will also profit from this change as MongoDB also generates non-integer ids.